ami/graphkit_wrapper.py

```diff
@@ -483,9 +488,26 @@ def __call__(self, *args, **kwargs):
         assert self.graphkit is not None, "call compile first"
         color = kwargs.get('color', None)
         assert color is not None
+        if color == 'globalCollector':
+            for node, names in self.latched_names.items():
+                inputs, outputs = names
+                if inputs.issubset(args[0].keys()):
+                    # print("RECOMPUTING:", outputs)
+                    continue
+                for output in outputs:
+                    if output in self.latch_cache:
+                        # print("LATCHING:", output)
+                        args[0][output] = self.latch_cache[output]
         result = self.graphkit(*args, **kwargs)
-        outputs = self.outputs[color]
-        return {k: result[k] for k in outputs if k in result}
+        if color == 'globalCollector':
+            for node, names in self.latched_names.items():
+                inputs, outputs = names
+                for output in outputs:
+                    if output in result:
+                        # print("UPDATING LATCH:", output)
+                        self.latch_cache[output] = result[output]
+        return {k: result[k] for k in self.outputs[color] if k in result}
+
```

Comment:
The main changes in this pull request are the lines above. When a blue box in the graph has been latched, on each heartbeat in the global collector we check if the local collector has input values for the blue box, if it does that triggers a recompute of the node on the global collector, if not then if we have previously computed the outputs of a global transformation on the global collector then we insert those values into the graph arguments so we can trigger a computation of downstream nodes from the global transformation. 

Then we update the cache with the new values.